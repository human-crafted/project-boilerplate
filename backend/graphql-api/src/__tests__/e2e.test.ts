import gql from 'graphql-tag';
import { server } from '..';
import { startTestServer, toPromise } from './test-utils';

const GET_USERS = gql`
  query userList {
    users {
      id
      firstName
      lastName
      emailAddress
      country
      signUpDate
    }
  }
`;

describe('Server - e2e', () => {
  let stop;
  let graphql;

  beforeEach(async () => {
    const testServer = await startTestServer(server);
    stop = testServer.stop;
    graphql = testServer.graphql;
  });

  afterEach(() => stop());

  it('Gets list of users', async () => {
    const res = await toPromise(
      graphql({
        query: GET_USERS
      })
    );
    expect(res).toMatchSnapshot();
  });
});
