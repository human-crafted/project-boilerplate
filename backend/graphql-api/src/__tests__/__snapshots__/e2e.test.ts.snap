// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server - e2e Gets list of users 1`] = `
Object {
  "data": Object {
    "users": Array [
      Object {
        "country": "GBR",
        "emailAddress": "leeroy.jenkins@humancrafted.tech",
        "firstName": "Leeroy",
        "id": "828382nsdjnj2",
        "lastName": "Jenkins",
        "signUpDate": "2020-01-20T16:29:56.6575020Z",
      },
    ],
  },
}
`;
